core/string/sub_spec.rb

```diff
@@ -396,7 +396,7 @@
   end
 
   it "ignores non-String keys" do
-    "hello".sub(/(ll)/, 'll' => 'r', :ll => 'z').should == "hero"
+    "hello".sub(/(ll)/, 'll' => 'r', ll: 'z').should == "hero"
```

Comment:
Numbers as symbols should be invalid
```
>> "hello".sub(/(11)/, '11' => 'r', 11: 'z')
SyntaxError: (irb):1: syntax error, unexpected ':', expecting =>
"hello".sub(/(11)/, '11' => 'r', 11: 'z')
```

The change here should have been:
```
>> "hello".sub(/(11)/, '11': 'r', 11 => 'z')
"hello"
```
With the quoted number having the syntax change, not the integer.